src/olympia/api/tests/test_pagination.py

```diff
@@ -54,36 +57,43 @@ def test_metadata_with_default_page_size(self):
 
 class TestESPageNumberPagination(TestCustomPageNumberPagination):
     def test_next_page_never_exeeds_max_result_window(self):
+        total = 50000
         mocked_qs = mock.MagicMock()
-        mocked_qs.__getitem__().execute().hits.total = 30000
+        mocked_qs.__getitem__().execute().hits.total = total
 
         view = generics.ListAPIView.as_view(
             serializer_class=PassThroughSerializer,
             queryset=mocked_qs,
             pagination_class=ESPageNumberPagination,
         )
 
-        request = self.factory.get('/', {'page_size': 5, 'page': 4999})
+        # Request the last page that still has a `next`.
+        page_size = 5
+        page = int(settings.ES_MAX_RESULT_WINDOW / page_size) - 1
+        request = self.factory.get('/', {'page_size': page_size, 'page': page})
         response = view(request)
         assert response.data == {
-            'page_size': 5,
-            'page_count': 5000,
+            'page_size': page_size,
+            'page_count': int(ceil(settings.ES_MAX_RESULT_WINDOW / page_size)),
```

Comment:
the discrepancy here makes me nervous - there's an extra `ceil` - shouldn't it be just `page + 1`?

Reply:
Or if there's some case I'm missing where there's a difference, we should have a test for that scenario?

Reply:
It was just to make the test generic and copy/pasteable

Reply:
committed another simplification in 89ee89001f692455fd71a4c50744afe52be2d1fc